### PR TITLE
Proper error reporting for assembly mode.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -921,13 +921,17 @@ bool CommandLineInterface::assemble()
...
 	for (auto const& stack: m_assemblyStacks)
+	{
 		for (auto const& error: stack.second.errors())
 			SourceReferenceFormatter::printExceptionInformation(
 				cerr,
 				*error,
 				(error->type() == Error::Type::Warning) ? "Warning" : "Error",
 				[&](string const& _source) -> Scanner const& { return *scanners.at(_source); }
 			);
+		if (!Error::containsOnlyWarnings(stack.second.errors()))
```
Comment (axic) [inline solc/CommandLineInterface.cpp:933]: Shouldn't `Assembly::assemble()` behave similarly to `CompilerStack::compiler()` and return false if non-warnings are emitted?

Comment (axic) [reply solc/CommandLineInterface.cpp:933]: Oh well, it returns a `LinkerObject` at the moment so it would need to return an empty/nil object.

Comment (chriseth) [reply solc/CommandLineInterface.cpp:933]: I thought about that, too, but I guess it's fine like that. We have a lot of calls to `containsOnlyWarnings` scattered around the codebase.

**Comment by chriseth** [comment]
Good to merge?

**Comment by axic** [comment]
> Good to merge?

I'm fine with the warning part, cannot really comment on the header shuffling.

**Review by axic — APPROVED** [review]
>Good to merge?

I'm fine with the warning part, cannot really comment on the header shuffling.

**Pull request merged**